### PR TITLE
Feat: add api of listing configs for project when creating a target

### PR DESCRIPTION
**Pull request by zzxwill**
When creating a target,  list configs by the config type and the project.

Signed-off-by: Zheng Xi Zhou <zzxwill@gmail.com>

### Description of your changes

Fixes #

I have:

- [ ] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

**Comment by zzxwill** [comment]
Verified by VeluaUX https://github.com/oam-dev/velaux/pull/429.

**Review by wonderflow — APPROVED** [review]

**Review by leejanee — APPROVED** [review]

File: pkg/apiserver/rest/usecase/project.go [hunk 1]
```diff
@@ -466,6 +471,40 @@ func (p *projectUsecaseImpl) UpdateProjectUser(ctx context.Context, projectName
 	return ConvertProjectUserModel2Base(&projectUser), nil
 }
 
+func (p *projectUsecaseImpl) GetConfigs(ctx context.Context, projectName, configType string) ([]*apisv1.Config, error) {
+	return getConfigsByProjectAndConfigType(ctx, p.k8sClient, projectName, configType)
+}
+
+func getConfigsByProjectAndConfigType(ctx context.Context, k8sClient client.Client, projectName, configType string) ([]*apisv1.Config, error) {
+	if configType != types.TerraformProvider {
```
Comment (barnettZQG) [inline pkg/apiserver/rest/usecase/project.go:479]: If only query TF providers, this function name should be getProjectTFProvider.
and the api path should be /api/v1/projects/xxx/tf_providers

Comment (zzxwill) [reply pkg/apiserver/rest/usecase/project.go:479]: We need to query the image registries list and chart repositories later.

File: pkg/apiserver/rest/webservice/project.go [hunk 2]
```diff
@@ -176,6 +176,15 @@ func (n *projectWebService) GetWebService() *restful.WebService {
 		Returns(200, "OK", []apis.PermissionBase{}).
 		Writes([]apis.PermissionBase{}))
 
+	ws.Route(ws.GET("/{projectName}/configs/").To(n.getConfigs).
```
Comment (barnettZQG) [inline pkg/apiserver/rest/webservice/project.go:179]:
```suggestion
	ws.Route(ws.GET("/{projectName}/configs").To(n.getConfigs).
```

File: pkg/apiserver/rest/webservice/project.go [hunk 3]
```diff
@@ -176,6 +176,15 @@ func (n *projectWebService) GetWebService() *restful.WebService {
...
+		Doc("get configs which are in a project").
+		Metadata(restfulspec.KeyOpenAPITags, tags).
+		Filter(n.rbacUsecase.CheckPerm("project", "list")).
+		Param(ws.PathParameter("projectName", "identifier of the project").DataType("string")).
+		Returns(200, "OK", []*apis.Config{}).
```
Comment (barnettZQG) [inline pkg/apiserver/rest/webservice/project.go:184]: add the query parameter definition.

File: pkg/apiserver/rest/usecase/project.go [hunk 4]
```diff
@@ -466,6 +472,94 @@ func (p *projectUsecaseImpl) UpdateProjectUser(ctx context.Context, projectName
...
+	var (
+		configs                  []*apisv1.Config
+		terraformProviders       []*apisv1.Config
+		legacyTerraformProviders []*apisv1.Config
+		apps                     = &v1beta1.ApplicationList{}
+	)
+	if err := p.k8sClient.List(ctx, apps, client.InNamespace(types.DefaultKubeVelaNS),
+		client.MatchingLabels{
+			model.LabelSourceOfTruth: model.FromInner,
+			types.LabelConfigCatalog: velaCoreConfig,
```
Comment (barnettZQG) [inline pkg/apiserver/rest/usecase/project.go:485]: filter by projectName?

Comment (zzxwill) [reply pkg/apiserver/rest/usecase/project.go:485]: No, `projectName` is optional when creating a config.

File: pkg/apiserver/rest/webservice/project.go (hunk 3 above)
Comment (zzxwill) [reply pkg/apiserver/rest/webservice/project.go:185]: Added.

**Review by barnettZQG — APPROVED** [review]
LGTM

**Pull request merged**